index.js

```diff
@@ -28,67 +28,75 @@ var cancelAnimFrame =
 	}
 ;
 
-// Class
-function AnimationLoop() {
-	this._drawMethods = [];
-	this._drawMethodsLen = 0;
-	this._isRunning = false;
-	this._requestId = null;
-}
+var drawMethods = [];
+var drawMethodsLen = 0;
+var isRunning = false;
+var requestId = null;
 
 // Add a method to run at every frame
-AnimationLoop.prototype.addDrawMethod = function (fn) {
+function addDrawMethod(fn) {
 	if (typeof fn === 'function') {
-		this._drawMethods.push(fn);
-		this._drawMethodsLen = this._drawMethods.length;
+		drawMethods.push(fn);
+		drawMethodsLen = drawMethods.length;
 	}
-};
+}
 
 // Remove all drawing methods
-AnimationLoop.prototype.removeAllDrawMethods = function () {
-	this._drawMethods.length = this._drawMethodsLen = 0;
-};
+function removeAllDrawMethods() {
+	drawMethods.length = drawMethodsLen = 0;
+}
 
 // Start the animation loop
-AnimationLoop.prototype.start = function () {
-	var that = this;
+function start() {
 	
 	if (anInstancesIsRunning) {
 		return console.error('AnimationLoop: impossible to start, another AnimationLoop instance is already running');
 	}
 	anInstancesIsRunning = true;
-	this._isRunning = true;
+	isRunning = true;
 	
 	function onTick() {
-		if (!that._isRunning) {
+		if (!isRunning) {
 			return;
 		}
-		if (that._drawMethodsLen === 0) {
-			return this.stop();
+		if (drawMethodsLen === 0) {
+			return stop();
 		}
 		
-		if (that._drawMethodsLen === 1) {
-			that._drawMethods[0]();
+		if (drawMethodsLen === 1) {
+			drawMethods[0]();
 		} else {
-			for (var i = 0; i < that._drawMethodsLen; i++) {
-				that._drawMethods[i]();
+			for (var i = 0; i < drawMethodsLen; i++) {
+				drawMethods[i]();
 			}
 		}
 		
-		that._requestId = requestAnimFrame(onTick);
+		requestId = requestAnimFrame(onTick);
 	}
 	
-	this._requestId = requestAnimFrame(onTick);
-};
+	requestId = requestAnimFrame(onTick);
+}
 
 // Stop the animation loop
-AnimationLoop.prototype.stop = function () {
-	this._isRunning = false;
+function stop() {
+	isRunning = false;
 	anInstancesIsRunning = false;
-	if (this._requestId) {
-		cancelAnimFrame(this._requestId);
+	if (requestId) {
+		cancelAnimFrame(requestId);
+	}
+}
+
+var instance;
+
+module.exports = function() {
+	if (!instance) {
+		return instance = {
+			addDrawMethod: addDrawMethod,
+			removeAllDrawMethods: removeAllDrawMethods,
+			start: start,
+			stop: stop
+		};
```

Comment:
@ronkorving I thought about it. Returning a whole constructor by default won't be any troubles and simple. But, to optionally requesting the instance re-use... there's a slight problem. See, say if some other components created their own instances and they are running independently no matter what. But, when a new component instantiates with "gimme the already running instance" option true, should we return the most recently created instance? See, it overwrites the private "instance" variable every time, and with the current design, we can only return the very latest. Yet, we never know he may want to re-use the 23rd instance created... then we need a feature to manage multiple instances.
It's nothing to do with the subject discussed, but I use RxJS to handle most of my ticking jobs nowadays, and find it very resource friendly. There will be only one observable stream by design, and all the worries go away because with observables, it is each module which manages what to do.
Anyways, I will work on the suggestion you made, and will make it backward compatible.